modules/test_infrastructure/variables.tf

```diff
@@ -0,0 +1,162 @@
+                                [VNET module documentation](../vnet/README.md#subnets)
+  - `network_security_groups` - (`map`, optional) map of Network Security Groups to create, for details see
+                                [VNET module documentation](../vnet/README.md#network_security_groups)
+  - `route_tables`            - (`map`, optional) map of Route Tables to create, for details see
+                                [VNET module documentation](../vnet/README.md#route_tables)
+  EOF
+  type = map(object({
+    name                    = string
+    resource_group_name     = optional(string)
+    create_virtual_network  = optional(bool, true)
+    address_space           = optional(list(string))
+    hub_resource_group_name = optional(string)
+    hub_vnet_name           = optional(string)
+    network_security_groups = optional(map(object({
+      name                          = string
+      disable_bgp_route_propagation = optional(bool)
+      rules = optional(map(object({
+        name                         = string
+        priority                     = number
+        direction                    = string
+        access                       = string
+        protocol                     = string
+        source_port_range            = optional(string)
+        source_port_ranges           = optional(list(string))
+        destination_port_range       = optional(string)
+        destination_port_ranges      = optional(list(string))
+        source_address_prefix        = optional(string)
+        source_address_prefixes      = optional(list(string))
+        destination_address_prefix   = optional(string)
+        destination_address_prefixes = optional(list(string))
+      })), {})
+    })), {})
+    route_tables = optional(map(object({
+      name = string
+      routes = map(object({
+        name                = string
+        address_prefix      = string
+        next_hop_type       = string
+        next_hop_ip_address = optional(string)
+      }))
+    })), {})
+    create_subnets = optional(bool, true)
+    subnets = optional(map(object({
+      name                            = string
+      address_prefixes                = optional(list(string), [])
+      network_security_group_key      = optional(string)
+      route_table_key                 = optional(string)
+      enable_storage_service_endpoint = optional(bool, false)
+    })), {})
+  }))
+}
+
+variable "hub_resource_group_name" {
+  description = "Name of the Resource Group hosting the hub/transit infrastructure. This value is required to create peering between the spoke and the hub VNET."
+  type        = string
+  default     = null
+}
+
+variable "hub_vnet_name" {
+  description = "Name of the hub/transit VNET. This value is required to create peering between the spoke and the hub VNET."
+  type        = string
+  default     = null
+}
+
+variable "test_vm_authentication" {
```

Comment:
how about simply `authentication`? This way you would follow the pattern for other modules. If you decide to parametrize the image configuration, an `image` variable would also follow the convention